src/dysh/spectra/tests/test_scan.py

```diff
@@ -183,6 +180,85 @@ def test_compare_with_GBTIDL(self, data_dir):
         # the difference at the ~0.06 K level
         assert np.nanmedian(ratio) <= 0.998
 
+    def test_synth_spectra(self, data_dir):
+        """Test subbeamnod using synthithic spectra."""
+        # Load a file with subbeamnod observations.
+        sdf_file = f"{data_dir}/TRCO_230413_Ka/TRCO_230413_Ka_scan43.fits"
+        sdf = gbtfitsload.GBTFITSLoad(sdf_file)
+
+        # Generate fake data.
+        def gauss(x, a, s, c):
+            return a * np.exp(-((x - c) ** 2.0) / (2 * s**2.0))
+
+        # Make a fake data set.
+        gain = 1e8
+        tsys = 100
+        tcont = 10
+        tcal = sdf["TCAL"][0]
+        a = 10
+        s = 5
+        c = 800
+        x = np.arange(0, sdf["DATA"].shape[1], 1)
+        nchan = sdf["DATA"].shape[1]
+
+        # Signal and reference integrations.
+        sig_mask = sdf["SUBREF_STATE"].to_numpy() == -1
+        ref_mask = sdf["SUBREF_STATE"].to_numpy() == 1
+        n_sig = sig_mask.sum()
+        n_ref = ref_mask.sum()
+
+        # Generate astro signal.
+        np.random.seed(0)  # Fix the noise.
+        sig_mod = np.random.normal(loc=0, scale=5, size=(n_sig, nchan)) + gauss(x, a, s, c) + tcont
+        ref_mod = np.random.normal(loc=0, scale=5, size=(n_ref, nchan))
+
+        # Add noise diode.
+        sig_cal_on = sdf["CAL"].to_numpy()[sig_mask] == "T"
+        sig_mod[sig_cal_on] += tcal
+        ref_cal_on = sdf["CAL"].to_numpy()[ref_mask] == "T"
+        ref_mod[ref_cal_on] += tcal
+
+        # Convert to counts.
+        p_sig_mod = gain * (sig_mod + tsys)
+        p_ref_mod = gain * (ref_mod + tsys)
+
+        # Replace data.
+        sdf["TCAL"] = tcal
+        new_data = np.empty_like(sdf["DATA"])
+        new_data[sig_mask] = p_sig_mod
+        new_data[ref_mask] = p_ref_mod
+        sdf["DATA"] = new_data
+
+        # Calibrate.
+        sbn_cycle = sdf.subbeamnod(scan=43, fdnum=0, plnum=0, ifnum=0).timeaverage()
+        sbn_scan = sdf.subbeamnod(scan=43, fdnum=0, plnum=0, method="scan").timeaverage()
+
+        # Check data over a frequency interval.
+        s_sbn = slice(30.0 * u.GHz, 30.5 * u.GHz)
+        rms_cycle = sbn_cycle[s_sbn].flux.std()
+        rms_scan = sbn_scan[s_sbn].flux.std()
+
+        # Compare continuum level.
+        assert pytest.approx(sbn_cycle.data.mean(), rms_cycle.value) == tcont
+        assert pytest.approx(sbn_scan.data.mean(), rms_scan.value) == tcont
+
+        # Compare exposure times.
+        assert sbn_cycle.meta["EXPOSURE"] == sbn_scan.meta["EXPOSURE"]
+
+        # Compare system temperature.
+        assert pytest.approx(sbn_scan.meta["TSYS"], rms_scan.value) == sbn_cycle.meta["TSYS"]
+        assert pytest.approx(sbn_cycle.meta["TSYS"] - tcal / 2.0, rms_cycle.value) == tsys
+        assert pytest.approx(sbn_scan.meta["TSYS"] - tcal / 2.0, rms_scan.value) == tsys
+        # The need to re-subtract the calibrator temperature suggests that there is a problem
```

Comment:
TPScan.calc_tsys() was re-written as part of #133.   Maybe a problem there?

Reply:
The base implementation of `spectrum.core.mean_tsys` did not change, and the remaining changes do not seem to affect the calculation itself, or other tests involving `gettp`. I think this may be a problem with the equation itself and adding back the `tcal/2.0` (in `spectrum.core.mean_tsys`), because the line amplitude also comes back higher than what you put in. I'd like to double check.

Reply:
Tsys computation reference: https://www.gb.nrao.edu/GBT/DA/gbtidl/gbtcal2.1-30Oct09.pdf

Reply:
Yes, not `dysh` specific.

Reply:
My mistake. Nothing wrong with how things are.